contracts/diaspore/DebtEngine.sol

```diff
@@ -205,7 +220,43 @@ contract DebtEngine is ERC721Base {
             _tokens: paidToken
         });
     }
-    
+
+    function payBatch(
+        bytes32[] _ids,
+        uint256[] _amounts,
+        address _oracle,
+        bytes8 _currency,
+        bytes _oracleData
+    ) external returns (uint256[], uint256[]) {
+        uint256 count = _ids.length;
+        require(count == _amounts.length, "The loans and the amounts do not correspond.");
+        require(count > 0, "There are not loans to pay.");
+
+        uint256[] memory data = new uint256[](3);
+        if (_oracle != address(0)) {
+            (data[1], data[2]) = IOracle(_oracle).getRate(_currency, _oracleData);
+            emit ReadedOracleBatch(count, data[1], data[2]);
+        }
+
+        uint256[] memory pays = new uint256[](count);
+        uint256[] memory paidTokens = new uint256[](count);
+        for (uint256 i = 0; i < count; i++) {
+            data[0] = _amounts[i];
+            (pays[i], paidTokens[i]) = _pay(_ids[i], _oracle, _currency, data);
```

Comment:
I am not sure if pays is a good choise of word, because we are mixing past (paid) and simple present (pays). Maybe I will use the sames variable names as pay() and payTokens() and have the plural implicit